packages/config/__snapshots__/validation.spec.ts.js

```diff
@@ -330,15 +330,13 @@ exports['extraneous keys'] = {
     'flaky': 5,
     'notflaky': null,
   },
-  'type': 'an object with keys `default` and `flaky`. Keys `default` and `flaky` must be integers greater than 0.',
+  'type': 'an object with keys `default` and `flaky`. Keys `default` and `flaky` must be integers between 0 and 250.',
 }
 
 exports['config/src/validation .isValidRetriesConfig experimental options fails with invalid strategy 1'] = {
-  'key': 'mockConfigKey',
-  'value': {
-    'experimentalStrategy': 'foo',
-  },
-  'type': 'a positive number or null or an object with keys "openMode" and "runMode" with values of numbers, booleans, or nulls, or experimental configuration with key "experimentalStrategy" with value "detect-flake-but-always-fail" or "detect-flake-and-pass-on-threshold" and key "experimentalOptions" to provide a valid configuration for your selected strategy',
+  'key': 'mockConfigKey.experimentalStrategy',
+  'value': 'foo',
+  'type': 'one of "detect-flake-but-always-fail", "detect-flake-and-pass-on-threshold"',
```

Comment:
These errors on config validation are way more readable and thank you for cleaning them up!

The one thing that is a bit critical here though is making sure that defaults get applied into the config so the app can access them through the client/server, which was one of the reasons it was an 'all or nothing' approach originally. For example:

```js
openMode: true,
experimentalStrategy: 'detect-flake-but-always-fail'
```

Is a valid end user config, but we need to add defaults to the object in the app so they can be globally available. This needs to be transformed into:

```js
openMode: true,
runMode: false,
experimentalStrategy: 'detect-flake-but-always-fail'.
experimentalOptions: {
   maxRetries: 2,
   stopIfAnyPassed: false,
}
```

`defaultValue` can take a function, but it doesn't expose the values present in the config at time of invocation.

The only thing I can think to accomplish this without adding a new mechanism, is to add defaults in the object when validation occurs since these happen by reference.